Makefile

```diff
@@ -3,6 +3,10 @@ default: build
 build:
 	go build -o terraform-provider-buildkite -ldflags="-s -w -X main.version=$(shell git describe --tag)" .
 
+.PHONY: build-snapshot
+build-snapshot:
+	goreleaser build --snapshot --clean --single-target
```

Comment:
I think there is value in keeping this building for all targets just in case there happens to be some issue (I think with go its highly unlikely at the compile stage at least though).

To improve CI speed, we can split the build out into parallel steps for each OS type (at some later stage)

Reply:
@jradtilbrook yeah I considered this, but just the sheer number of architectures kind of scared me off.

But lets try it.

Reply:
Oh yeah thats okay. Save that for another time!